lib/board.rb

```diff
@@ -0,0 +1,46 @@
+class Board
+  attr_accessor :cells
+
+  def reset!
+    @cells.clear
+    @cells = Array.new(9, " ")
+  end
+
+  def initialize
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(take)
+    @cells[take.to_i - 1]
+  end
+
+  def full?
+    (@cells.all? { |turn| (turn == "X" || turn == "O") }) ? true : false
+  end
+
+  def turn_count
+    counter = 0
+    @cells.each { |turn| counter += 1 if turn == "X" || turn == "O" }
+    counter
+  end
+
+  def taken?(index)
+    (@cells[index.to_i - 1] == " " || nil) ? false : true
+  end
+
+  def valid_move?(index)
+    (index.to_i.between?(1, 9) && (@cells[index.to_i - 1] == " " || @cells[index.to_i - 1] == nil)) ? true : false
```

Comment:
a ternary operator is not needed for an expression statement like this. The implicit return value for an expression is true or false so if both sides of the && statement return true `valid_move?` returns true else false.